openbas-api/src/main/java/io/openbas/service/ScenarioToExerciseService.java

```diff
@@ -24,262 +24,266 @@
-    private final ArticleRepository articleRepository;
-    private final LessonsCategoryRepository lessonsCategoryRepository;
-    private final LessonsQuestionRepository lessonsQuestionRepository;
-    private final InjectRepository injectRepository;
-    private final InjectDocumentRepository injectDocumentRepository;
-    private final VariableService variableService;
-    private final TeamService teamService;
-    @Resource
-    protected ObjectMapper mapper;
+  private final ExerciseRepository exerciseRepository;
+  private final GrantRepository grantRepository;
+  private final TeamRepository teamRepository;
+  private final ExerciseTeamUserRepository exerciseTeamUserRepository;
+  private final ObjectiveRepository objectiveRepository;
+  private final DocumentRepository documentRepository;
+  private final ArticleRepository articleRepository;
+  private final LessonsCategoryRepository lessonsCategoryRepository;
+  private final LessonsQuestionRepository lessonsQuestionRepository;
+  private final InjectRepository injectRepository;
+  private final InjectDocumentRepository injectDocumentRepository;
+  private final VariableService variableService;
+  private final TeamService teamService;
+  @Resource
+  protected ObjectMapper mapper;
 
-    @Transactional(rollbackFor = Exception.class)
-    public Exercise toExercise(
-            @NotBlank final Scenario scenario,
-            @Nullable final Instant start) {
-        Exercise exercise = new Exercise();
-        exercise.setScenario(scenario);
-        exercise.setName(scenario.getName());
-        exercise.setDescription(scenario.getDescription());
-        exercise.setSubtitle(scenario.getSubtitle());
-        exercise.setCategory(scenario.getCategory());
-        exercise.setMainFocus(scenario.getMainFocus());
-        exercise.setSeverity(scenario.getSeverity());
-        exercise.setHeader(scenario.getHeader());
-        exercise.setFooter(scenario.getFooter());
-        exercise.setFrom(scenario.getFrom());
-        exercise.addReplyTos(scenario.getReplyTos());
-        exercise.setStart(start);
+  @Transactional(rollbackFor = Exception.class)
+  public Exercise toExercise(
+      @NotBlank final Scenario scenario,
+      @Nullable final Instant start,
+      @Nullable final Boolean isRunning) {
+    Exercise exercise = new Exercise();
+    exercise.setScenario(scenario);
+    exercise.setName(scenario.getName());
+    exercise.setDescription(scenario.getDescription());
+    exercise.setSubtitle(scenario.getSubtitle());
+    exercise.setCategory(scenario.getCategory());
+    exercise.setMainFocus(scenario.getMainFocus());
+    exercise.setSeverity(scenario.getSeverity());
+    exercise.setHeader(scenario.getHeader());
+    exercise.setFooter(scenario.getFooter());
+    exercise.setFrom(scenario.getFrom());
+    exercise.addReplyTos(scenario.getReplyTos());
+    exercise.setStart(start);
+    if (isRunning != null && isRunning) {
+      exercise.setStatus(ExerciseStatus.RUNNING);
```

Comment:
If isRunning is a primitive boolean

```suggestion
    if (isRunning) {
      exercise.setStatus(ExerciseStatus.RUNNING);
```